### PR TITLE
feat(perf): Use metrics in transaction summary

### PR DESCRIPTION
**Pull request by udameli**
When query is compatible with metrics dataset, use it to power transaction summary. When it's not possible, use indexed event and show a warning explaining the change.
Screen Shot 2023-02-02 at 6 35 42 PM
Screen Shot 2023-02-02 at 6 36 04 PM
Screen Shot 2023-02-02 at 6 36 14 PM

### CODE REVIEW

File: static/app/views/performance/transactionSummary/transactionOverview/content.tsx [hunk 1]
```diff
@@ -498,4 +526,15 @@ const StyledSearchBar = styled(SearchBar)`
   }
 `;
 
+const StyledWarningIcon = styled(IconWarning)`
+  color: ${p => p.theme.alert.warning.iconColor};
+  height: ${p => p.theme.iconSizes.sm};
+  width: ${p => p.theme.iconSizes.sm};
```
Comment (evanpurkhiser) [inline static/app/views/performance/transactionSummary/transactionOverview/content.tsx:532]: I think you can set the size and color prop on the icon and you shouldn't need to style it

File: static/app/views/performance/transactionSummary/transactionOverview/content.tsx [hunk 2]
```diff
@@ -498,4 +526,15 @@ const StyledSearchBar = styled(SearchBar)`
...
+  display: block;
+
+  &:hover {
+    color: ${p => p.theme.alert.warning.iconHoverColor};
```
Comment (evanpurkhiser) [inline static/app/views/performance/transactionSummary/transactionOverview/content.tsx:536]: Does it need a hover?

(I think you can probably delete this styled component)

File: static/app/views/performance/transactionSummary/transactionOverview/userStats.tsx [hunk 3]
```diff
@@ -44,9 +44,11 @@ function UserStats({
   let userMisery = error !== null ? <div>{'\u2014'}</div> : <Placeholder height="34px" />;
 
   if (!isLoading && error === null && totals) {
-    const threshold: number | undefined = totals.project_threshold_config[1];
+    const threshold: number | undefined = totals.project_threshold_config
+      ? totals.project_threshold_config[1]
+      : undefined;
     const miserableUsers: number | undefined = totals['count_miserable_user()'];
-    const userMiseryScore: number = totals['user_misery()'];
+    const userMiseryScore: number = totals['user_misery()'] | 0;
```
Comment (k-fish) [inline static/app/views/performance/transactionSummary/transactionOverview/userStats.tsx:51]: bitwise OR intentional here?

File: static/app/views/performance/transactionSummary/transactionOverview/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  const display = decodeScalar(
+    location.query.display,
+    DisplayModes.DURATION
+  ) as DisplayModes;
+  const breakdown = decodeScalar(location.query.breakdown, '');
+  const query = decodeScalar(location.query.query, '');
+
+  // certain charts aren't compatible with metrics
+  if (DISPLAY_MAP_DENY_LIST.includes(display)) {
```
Comment (k-fish) [inline static/app/views/performance/transactionSummary/transactionOverview/utils.tsx:29]: So all future views work on processed unless explicitly denied? 👍

**Review by k-fish — APPROVED** [review]
Maybe fix the bitwise OR, otherwise 👍

File: static/app/views/performance/transactionSummary/transactionOverview/userStats.tsx (hunk 3 above)
Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionOverview/userStats.tsx:51]: oops, definitely not intentional

File: static/app/views/performance/transactionSummary/transactionOverview/content.tsx (hunk 1 above)
Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionOverview/content.tsx]: ah nice, kk added 👍

File: static/app/views/performance/transactionSummary/transactionOverview/content.tsx (hunk 2 above)
Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionOverview/content.tsx]: I needed to center it so I still needed to apply` display` attribute unfortunately

**Pull request merged**